CHANGELOG.md

```diff
@@ -11,6 +11,112 @@ All notable changes to Botonic will be documented in this file.
   </summary>
 </details>
 
+## [0.12.0] - 2020-26-05
+
+### Added
+
+- [Project](https://github.com/hubtype/botonic)
+
+  - Added [ESLint Limit Cyclomatic Complexity](https://eslint.org/docs/rules/complexity) for all projects.
+  - Added Pull Request Templates.
+  - Scripts to count lines of code on each package.
+  - Added new documentation built with `Docusaurus`. Check it out at https://docs.botonic.io/.
+
+* [@botonic/cli](https://www.npmjs.com/package/@botonic/cli)
+
+  - Enable more checks in `tsconfig.json`.
+
+- [@botonic/core](https://www.npmjs.com/package/@botonic/core)
+
+  - Allow transferring conversations to an agent with its identifier (id or email) with `withAgentId(agentId)` or `withAgentEmail(agentEmail)` method of `HandOffBuilder`. Now `queue`is not mandatory. The case will be assigned to the first queue which agent belongs.
+  - Added `getAgentVacationRanges` to know availability of agents.
+  - Added more `router`tests.
+  - Type definitions improvement in `index.d.ts`.
+
+* [@botonic/react](https://www.npmjs.com/package/@botonic/react)
+
+  - General features:
+
+    - Type definitions improvement in `index.d.ts`.
+    - Improved `Text.serialize` method.
+
+  - New webchat features:
+
+    - **Breaking change**: Now the text of a `Button/Reply` is sent along with its payload. You can use `button: {messageType: 'postback'}` in `webchat/index.js` to enable the previous behavior.
+    - **Breaking change**: Markdown will be rendered by default within all the `Text` components. If you want to disable it, you can declare the component as follows:  
+      `<Text markdown={false}>Your text</Text>`.
+    - Be able to pass additional styles to `Custom Messages`.
+    - Persistent Menu customizable with property `customPersistentMenu` in `webchat/index.js`.
+    - Persistent Menu button customizable with property `customMenuButton`.
+    - Possibility to enable timestamps in messages.
+    - Added unit tests for Webchat.
+    - More adaptable Webchat in iOS.
+
+  - Multichannel:
+
+    - Customize `Multichannel` visualization.
+    - `Multichannel` to support letter indexes.
+    - Compact `Multichannel` `Text` and `Carousel`.
+    - `Multichannel` to allow separating messages with custom string.
+
+- [@botonic/plugin-contentful](https://www.npmjs.com/package/@botonic/plugin-contentful)
+  - `Comment` and `Chitchat` typing improvements.
+  - Allow passing `Normalizer` to `CMS`.
+  - Added Markup (WhatsApp Support) and Markdown parsing.
+  - Added stemmers/tokens for Russian.
+
```

Comment:
- Now normalizer throws an exception for empty texts.
- Keyword search now sets the result's score field